jar-submission-module/jar-submission-test/src/main/java/com/hazelcast/jet/tests/jarsubmission/JarSubmissionTest.java

```diff
@@ -0,0 +1,349 @@
+
+                // check result
+                IMap<String, String> map = client.getMap(prefix);
+                if (jobCount % 10 == 0) {
+                    assertJobStatusEventually(job, JobStatus.FAILED);
+                    assertNotEquals(prefix + jobCount, map.get(prefix));
+                } else {
+                    assertJobStatusEventually(job, JobStatus.RUNNING);
+                    assertMapItemEventually(map, prefix, prefix + jobCount);
+                    job.cancel();
+                }
+            }
+
+            if (jobCount % logThreshold == 0) {
+                logger.info("Job count for " + prefix + ": " + jobCount);
+            }
+            jobCount++;
+            sleepMillis(sleep);
+        }
+        assertTrue(jobCount > 0);
+        logger.info("Job count for " + prefix + ": " + jobCount);
+    }
+
+    private static Job getJobEventually(JetService jet, String jobName) {
+        for (int i = 0; i < ASSERTION_ATTEMPS; i++) {
+            Job job = jet.getJob(jobName);
+            if (job != null) {
+                return job;
+            } else {
+                sleepMillis(ASSERTION_SLEEP_MS);
+            }
+        }
+        throw new AssertionError("Job " + jobName + " was not found.");
+    }
+
+    private static void assertMapItemEventually(IMap<String, String> map, String key, String expectedValue) {
+        for (int i = 0; i < ASSERTION_ATTEMPS; i++) {
+            String value = map.get(key);
+            if (value != null && value.equals(expectedValue)) {
+                return;
+            } else {
+                sleepMillis(ASSERTION_SLEEP_MS);
+            }
+        }
+        throw new AssertionError("Map " + map.getName() + " does not contain expected value for key: " + key
+                + ". Expected value: " + expectedValue + " actual value: " + map.get(key));
+    }
+
+    private static void assertJobStatusEventually(Job job, JobStatus expectedStatus) {
+        for (int i = 0; i < ASSERTION_ATTEMPS; i++) {
+            if (job.getStatus().equals(expectedStatus)) {
+                return;
+            } else {
+                sleepMillis(ASSERTION_SLEEP_MS);
+            }
+        }
+        throw new AssertionError("Job " + job.getName() + " does not have expected status: " + expectedStatus
+                + ". Job status: " + job.getStatus());
+    }
+
+    private void createLargeJar() throws IOException {
+        generateLargeFile();
+        Files.copy(Paths.get(smallJarPath), Paths.get(largeJarPath), StandardCopyOption.REPLACE_EXISTING);
+
+        Map<String, String> env = new HashMap<>();
+        env.put("create", "true");
+        URI uri = URI.create("jar:" + Paths.get(largeJarPath).toUri());
+        try (FileSystem fs = FileSystems.newFileSystem(uri, env)) {
+            Path nf = fs.getPath("large-file.csv");
+            Files.write(nf, Files.readAllBytes(Paths.get(largeFilePath)), StandardOpenOption.CREATE);
```

Comment:
I thought about that but I decided to use this approach because we can reuse it if we will decide that we would like to try to send for example file with 4G for whatever reason (then client memory would not be sufficient).